### DISCUSSION TITLE
Randomly getting 404s from Vercel Serverless Functions with Express.js API

### DISCUSSION THREAD

**Discussion by OwaisSiddiqui** [vercel/community#3598] [Help] [answered]
### Summary

GitHub of project deployed to Vercel: https://github.com/OwaisSiddiqui/flags-online

I have a React frontend and Express.js API backend (with tRPC) app setup. I have deployed it to Vercel. The Express.js API is on Vercel Serverless Functions. The app works perfectly fine on development with no errors.

The problem I'm having is that it will randomly return 404s when accessing an API route. For example, when I go on the home page right now, https://flags-online.com/", when a user inputs a username for sign up, the app first checks if the username exists by calling the API route isUserExist: https://flags-online.com/api/trpc/user.isUserExist?batch=1d. I input a username and it does nothing. I check the Chrome Dev console and it is a 404 error in the Network tab "Cannot POST /api/trpc/user.isUserExist". 

Failed request:

fetch("https://flags-online.com/api/trpc/user.isUserExist?batch=1", {
  "headers": {
    "authorization": "Bearer null",
    "content-type": "application/json",
    "sec-ch-ua": "\"Not/A)Brand\";v=\"99\", \"Google Chrome\";v=\"115\", \"Chromium\";v=\"115\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\""
  },
  "referrer": "https://flags-online.com/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": "{\"0\":{\"username\":\"ceo\"}}",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});

The weird thing is that I will just try it again right after and it would start working. 

Successful request:

fetch("https://flags-online.com/api/trpc/user.isUserExist?batch=1", {
  "headers": {
    "authorization": "Bearer null",
    "content-type": "application/json",
    "sec-ch-ua": "\"Not/A)Brand\";v=\"99\", \"Google Chrome\";v=\"115\", \"Chromium\";v=\"115\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"Windows\""
  },
  "referrer": "https://flags-online.com/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": "{\"0\":{\"username\":\"ceo\"}}",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});

Response:

[
    {
        "result": {
            "data": true
        }
    }
]

This happens throughout the app with different API routes. How can I fix this?

### Example

https://github.com/OwaisSiddiqui/flags-online

### Steps to Reproduce

1. Go to https://flags-online.com/.
2. Input a username and tab
3. See Network tab and check that a 404 error occurred when trying to access API

**Comment by Romannoa** [comment]
hm so hard

**Comment by OwaisSiddiqui** [comment, marked as answer]
Error on calling initDb from server directly in api/index.ts file, resolved by initDb for each request
